metadata/logger.go

```diff
@@ -11,14 +11,37 @@
 
 package metadata
 
-import "github.com/go-logr/logr"
+var log Logger = DiscardLogger{}
 
-var log logr.Logger = logr.Discard()
+// Logger partially implements the go-log/logr's interface:
+// https://github.com/go-logr/logr/blob/master/logr.go
+type Logger interface {
+	// Info logs a non-error message with key/value pairs
+	Info(msg string, kv ...any)
+	// Error logs an error with a given message and key/value pairs.
+	Error(err error, msg string, kv ...any)
+	// V returns a new logger with updated log level.
+	V(level int) Logger
+}
+
+type DiscardLogger struct{}
+
+func (d DiscardLogger) Info(msg string, kv ...any) {
+
+}
+
+func (d DiscardLogger) Error(err error, msg string, kv ...any) {
+
+}
+
+func (d DiscardLogger) V(l int) Logger {
```

Comment:
I don't really like this part of the interface, as it becomes hard for an implementation to satisfy this. See `examples/logger` for an example of a simple wrapper.